### PR TITLE
Correct typos in Quicksight flatteners for `visible_range`,  `rows`, and `visibility`

### LINKED ISSUE #31923: [Bug]: Error setting definition in aws_quicksight_template
### Terraform Core Version

1.3.4

### AWS Provider Version

5.3.0

### Affected Resource(s)

aws_quicksight_template
aws_quicksight_dashboard

### Expected Behavior

The template should be update according to the source analysis.

### Actual Behavior

Terraform throws an error while it's trying to set the definition.
This error occurs on multiple different addresses. 

### Relevant Error/Panic Output Snippet

```shell
Error: setting definition: Invalid address to set: []string{"definition", "0", "sheets", "0", "visuals", "0", "bar_chart_visual", "0", "chart_configuration", "0", "category_axis", "0", "scrollbar_options", "0", "visibile_range"}

Other errors:
Error: setting definition: Invalid address to set: []string{"definition", "0", "sheets", "0", "visuals", "0", "pivot_table_visual", "0", "chart_configuration", "0", "field_options", "0", "selected_field_options", "0", "visbility"}
Error: setting definition: Invalid address to set: []string{"definition", "0", "sheets", "0", "visuals", "0", "pivot_table_visual", "0", "chart_configuration", "0", "field_wells", "0", "pivot_table_aggregated_field_wells", "0", "row"}
```

### Terraform Configuration Files

```terraform
provider "aws" {
  region = "eu-west-1"
}

terraform {
  required_version = "~> 1.3.4"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.3.0"
    }
  }
}

resource "aws_quicksight_template" "this" {
  template_id         = "template"
  name                = "Template"
  version_description = timestamp()

  definition {
    data_set_configuration {
      placeholder = "placeholder"
      data_set_schema {
        column_schema_list {
          name      = "column1"
          data_type = "STRING"
        }
        column_schema_list {
          name      = "column2"
          data_type = "STRING"
        }
      }
    }
    sheets {
      sheet_id = "sheet1"
      visuals {
        bar_chart_visual {
          visual_id = "visual1"
          chart_configuration {
            field_wells {
              bar_chart_aggregated_field_wells {
                category {
                  categorical_dimension_field {
                    field_id = "category-id"
                    column {
                      data_set_identifier = "placeholder"
                      column_name         = "column1"
                    }
                  }
                }
                colors {
                  categorical_dimension_field {
                    field_id = "color-id"
                    column {
                      data_set_identifier = "placeholder"
                      column_name         = "column2"
                    }
                  }
                }
              }
            }
            category_axis {
              scrollbar_options {
                visible_range {
                  percent_range {
                    from = "25.5"
                    to   = "36.7"
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

### Steps to Reproduce

terraform init
terraform apply

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

The same behavior occurs when a template is created through a definition block or when it's created from a source analysis.

### References

_No response_

### Would you like to implement a fix?

None

### LINKED ISSUE #32347: [Bug]: Cannot set visible_range of `bar_chart_visual` in a quicksight analysis
### Terraform Core Version

1.5.2

### AWS Provider Version

5.5.0

### Affected Resource(s)

aws_quicksight_analysis

### Expected Behavior

Setting the scrollbar options of a bar chart visual to a percentage range of 0-100 to be honoured in the analysis

### Actual Behavior

An exception is thrown when creating the analysis

### Relevant Error/Panic Output Snippet

```shell
╷
│ Error: setting definition: Invalid address to set: []string{"definition", "0", "sheets", "0", "visuals", "0", "bar_chart_visual", "0", "chart_configuration", "0", "category_axis", "0", "scrollbar_options", "0", "visibile_range"}
│
```

### Terraform Configuration Files

```
resource "aws_quicksight_analysis" "example_download_analysis" {
  analysis_id    = "example-distribution-analysis"
  name           = "Example Analysis"
  aws_account_id = var.account_id

  definition {
    data_set_identifiers_declarations {
      data_set_arn = var.dataset_arn
      identifier   = var.dataset_id
    }

    sheets {
      sheet_id = "example-sheet"
      name     = "Example"

      visuals {
        bar_chart_visual {
          visual_id = "failed-download-bar-chart"
          title {
            visibility = "VISIBLE"
            format_text {
              rich_text = "<visual-title>Failed Download HTTP Statuses</visual-title>"
            }
          }
          subtitle {
            visibility = "VISIBLE"
          }
          chart_configuration {
            field_wells {
              bar_chart_aggregated_field_wells {
                category {
                  date_dimension_field {
                    field_id = "failed-download-time-field"
                    column {
                      data_set_identifier = var.dataset_id
                      column_name         = "time"
                    }
                    hierarchy_id = "failed-download-time-field"
                  }
                }
                colors {
                  categorical_dimension_field {
                    field_id = "failed-download-status-field"
                    column {
                      data_set_identifier = var.dataset_id
                      column_name         = "http_status"
                    }
                  }
                }
              }
            }
            sort_configuration {
              category_sort {
                field_sort {
                  field_id  = "failed-download-time-field"
                  direction = "ASC"
                }
              }
              category_items_limit {
                items_limit      = 5
                other_categories = "INCLUDE"
              }
              color_items_limit {
                items_limit      = 5
                other_categories = "INCLUDE"
              }
              small_multiples_limit_configuration {
                other_categories = "INCLUDE"
                items_limit      = 1
              }
            }
            orientation      = "VERTICAL"
            bars_arrangement = "CLUSTERED"
            category_axis {
              scrollbar_options {
                visible_range {
                  percent_range {
                    from = 0
                    to = 100
                  }
                }
              }
            }
            color_label_options {
              sort_icon_visibility = "VISIBLE"
            }
            legend {
              width = "108px"
            }
            data_labels {
              visibility = "HIDDEN"
              overlap    = "DISABLE_OVERLAP"
            }
            tooltip {
              tooltip_visibility    = "VISIBLE"
              selected_tooltip_type = "BASIC"
              field_base_tooltip {
                aggregation_visibility = "HIDDEN"
                tooltip_title_type     = "PRIMARY_VALUE"
                tooltip_fields {
                  field_tooltip_item {
                    field_id   = "failed-download-time-field"
                    visibility = "VISIBLE"
                  }
                }
                tooltip_fields {
                  field_tooltip_item {
                    field_id   = "failed-download-status-field"
                    visibility = "VISIBLE"
                  }
                }
              }
            }
          }
          column_hierarchies {
            date_time_hierarchy {
              hierarchy_id = "failed-download-time-field"
            }
          }
        }
      }

      layouts {
        configuration {
          grid_layout {
            elements {
              element_id   = "failed-download-bar-chart"
              element_type = "VISUAL"
              column_index = 0
              column_span  = 13
              row_index    = 0
              row_span     = 12
            }
            canvas_size_options {
              screen_canvas_size_options {
                resize_option             = "FIXED"
                optimized_view_port_width = "1600px"
              }
            }
          }
        }
      }

      content_type = "INTERACTIVE"
    }

    analysis_defaults {
      default_new_sheet_configuration {
        interactive_layout_configuration {
          grid {
            canvas_size_options {
              screen_canvas_size_options {
                resize_option             = "FIXED"
                optimized_view_port_width = "1600px"
              }
            }
          }
        }
        sheet_content_type = "INTERACTIVE"
      }
    }
  }

  permissions {
    principal = var.quicksight_data_group
    actions = [
      "quicksight:DeleteAnalysis",
      "quicksight:DescribeAnalysis",
      "quicksight:DescribeAnalysisPermissions",
      "quicksight:QueryAnalysis",
      "quicksight:RestoreAnalysis",
      "quicksight:UpdateAnalysis",
      "quicksight:UpdateAnalysisPermissions",
    ]
  }
}
```

### Steps to Reproduce

Create a bar chart visual in a quicksight analysis and apply changes to AWS

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by justinretzolk**
### Description

There were three typos in some of the Quicksight flattener functions that were leading to the following errors. This PR aims to correct them so that the data can successfully be written to the state.

I'm unsure on running the acceptance tests for this, so have left that out for now. I'd also be interested in how to properly approach the changelog entry for this, since it affects multiple resources due to the shared schema.

```shell
Error: setting definition: Invalid address to set: []string{"definition", "0", "sheets", "0", "visuals", "0", "bar_chart_visual", "0", "chart_configuration", "0", "category_axis", "0", "scrollbar_options", "0", "visibile_range"}
Error: setting definition: Invalid address to set: []string{"definition", "0", "sheets", "0", "visuals", "0", "pivot_table_visual", "0", "chart_configuration", "0", "field_options", "0", "selected_field_options", "0", "visbility"}
Error: setting definition: Invalid address to set: []string{"definition", "0", "sheets", "0", "visuals", "0", "pivot_table_visual", "0", "chart_configuration", "0", "field_wells", "0", "pivot_table_aggregated_field_wells", "0", "row"}
```

### Relations

Closes #31923
Closes #32347

### References

Schema reference for:

- [`visible_range`](https://github.com/hashicorp/terraform-provider-aws/blob/a2405f6611d18c0644d8c5ac967f1eb04bb5171f/internal/service/quicksight/schema/visual_chart_configuration.go#L147)
- [`rows`](https://github.com/hashicorp/terraform-provider-aws/blob/a2405f6611d18c0644d8c5ac967f1eb04bb5171f/internal/service/quicksight/schema/visual_pivot_table.go#L98)
- [`visibility`](https://github.com/hashicorp/terraform-provider-aws/blob/a2405f6611d18c0644d8c5ac967f1eb04bb5171f/internal/service/quicksight/schema/visual_pivot_table.go#L76)

### Output from Acceptance Testing

The acceptance tests for Quicksight are a bit unclear to me, so I'll need some assistance in running those, should it be needed for these typo fixes.

### CODE REVIEW

**Review by jar-b — APPROVED** [review]
LGTM 🎉 


```console
$ make testacc PKG=quicksight TESTS=TestAccQuickSightAnalysis_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/quicksight/... -v -count 1 -parallel 20 -run='TestAccQuickSightAnalysis_'  -timeout 180m
?       github.com/hashicorp/terraform-provider-aws/internal/service/quicksight/schema  [no test files]

--- PASS: TestAccQuickSightAnalysis_forceDelete (66.13s)
--- PASS: TestAccQuickSightAnalysis_parametersConfig (77.63s)
--- PASS: TestAccQuickSightAnalysis_disappears (82.51s)
--- PASS: TestAccQuickSightAnalysis_sourceEntity (82.62s)
--- PASS: TestAccQuickSightAnalysis_basic (88.76s)
--- PASS: TestAccQuickSightAnalysis_update (123.69s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/quicksight 127.049s
```

```console
$ make testacc PKG=quicksight TESTS=TestAccQuickSightDashboard_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/quicksight/... -v -count 1 -parallel 20 -run='TestAccQuickSightDashboard_'  -timeout 180m
?       github.com/hashicorp/terraform-provider-aws/internal/service/quicksight/schema  [no test files]

--- PASS: TestAccQuickSightDashboard_dashboardSpecificConfig (74.58s)
--- PASS: TestAccQuickSightDashboard_basic (75.44s)
--- PASS: TestAccQuickSightDashboard_disappears (82.13s)
--- PASS: TestAccQuickSightDashboard_sourceEntity (87.14s)
--- PASS: TestAccQuickSightDashboard_updateVersionNumber (130.51s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/quicksight 133.963s
```

```console
$ make testacc PKG=quicksight TESTS=TestAccQuickSightTemplate_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/quicksight/... -v -count 1 -parallel 20 -run='TestAccQuickSightTemplate_'  -timeout 180m
?       github.com/hashicorp/terraform-provider-aws/internal/service/quicksight/schema  [no test files]

--- PASS: TestAccQuickSightTemplate_disappears (56.96s)
--- PASS: TestAccQuickSightTemplate_barChart (61.01s)
--- PASS: TestAccQuickSightTemplate_basic (61.47s)
--- PASS: TestAccQuickSightTemplate_sourceEntity (63.19s)
--- PASS: TestAccQuickSightTemplate_table (87.59s)
--- PASS: TestAccQuickSightTemplate_update (96.64s)
--- PASS: TestAccQuickSightTemplate_tags (114.31s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/quicksight 117.657s
```

File: .changelog/32464.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+```release-note:bug
+resource/aws_quicksight_*: Fix exception thrown when specifying `definition.sheets.visuals.bar_chart_visual.chart_configuration.category_axis.scrollbar_options.visible_range`
```
Comment (jar-b) [inline .changelog/32464.txt:2]: If you'd like to break these out into explicit resources, the three which share this underlying schema are `aws_quicksight_template`, `aws_quicksight_dashboard`, and `aws_quicksight_analysis`.

Comment (justinretzolk) [reply .changelog/32464.txt]: Perfect, thanks! I've pushed that change.

**Review by jar-b — APPROVED** [review]
LGTM 👍

**Pull request merged**